pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -263,8 +223,17 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	if mapCtx.Err() != nil {
 		return nil, mapCtx.Err()
 	}
-	obj.Spec.ProjectRef = &refs.ProjectRef{Name: a.id.Parent.Project}
-	obj.Spec.Location = a.id.Parent.Location
+	parent, err := a.id.Parent()
+	if err != nil {
+		return nil, fmt.Errorf("BigQueryConnectionConnection %s parent unset: %w", a.id.External, err)
+	}
+	if parent != "" {
+		tokens := strings.Split(parent, "/")
+		if len(tokens) == 4 && tokens[0] == "projects" && tokens[2] == "locations" {
```

Comment:
Interesting ... the Parent is the last thing here that I think is awkward, and even now Parent still isn't exactly what we need for export.  Let's see how this evolves!